### PR TITLE
[JBPM-9970] Test for Process instance migration does not update command data

### PR DESCRIPTION
**Pull request by gmunozfe**
**JIRA**: [JBPM-9970](https://issues.redhat.com/browse/JBPM-9970)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* paste the link(s) from GitHub here
* link 2
* link 3 etc.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by gmunozfe** [comment]
jenkins retest this

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/JobServiceMigrationIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        assertTrue(errors.size() > 0);
+        
+        Long jobId = errors.get(0).getId();
+        
+        // migrate process instance to container 2 (right deployment-structure containing the CustomWorkItemHandler definition)
+        MigrationReportInstance report = processAdminClient.migrateProcessInstance(CONTAINER_ID1, pid, CONTAINER_ID2, PROCESS_ID);
+        assertNotNull(report);
+        assertTrue(report.isSuccessful());
+        
+        jobServicesClient.updateRequestData(jobId, null, new HashMap<>());
```
Comment (afalhambra) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/JobServiceMigrationIntegrationTest.java:88]: why do we need to update the request data? is it needed?

**Review by afalhambra — COMMENTED** [review]
Looks great to me, just a couple of question. Thanks @gmunozfe!

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/JobServiceMigrationIntegrationTest.java (hunk 1 above)
Comment (gmunozfe) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/JobServiceMigrationIntegrationTest.java:88]: Not really needed, removed, thanks for realizing @afalhambra

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/JobServiceMigrationIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        jobServicesClient.requeueRequest(jobId);
+        
+        KieServerSynchronization.waitForJobToFinish(jobServicesClient, jobId);
+        
+        RequestInfoInstance jobRequest = jobServicesClient.getRequestById(jobId, false, false);
```
[the remark this answers is not in the document]
Comment (gmunozfe) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/JobServiceMigrationIntegrationTest.java:93]: That's just to check the job is done, no matter the request is with or without data.

**Review by afalhambra — APPROVED** [review]
Nice one!

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
jenkins retest this

**Review by afalhambra — APPROVED** [review]
Great job!

**Comment by gmunozfe** [comment]
@mareknovotny ready to merge

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/JobServiceMigrationIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        long id = jsc.scheduleRequest(JobRequestInstance.builder().command("org.jbpm.executor.commands.ExecutionErrorCleanupCommand").build());
+        KieServerSynchronization.waitForJobToFinish(jsc, id, 120000L);
+    }
+    
+    @Test
+    public void testMigrateAndRequeueFailingJob() throws Exception {
+        Long pid = processClient.startProcess(CONTAINER_ID1, PROCESS_ID);
+        
+        // short wait before querying the async job id
+        Thread.sleep(2000);
```
Comment (mareknovotny) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/JobServiceMigrationIntegrationTest.java:83]: This way is not optimal for running on different machines.

Comment (mareknovotny) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/JobServiceMigrationIntegrationTest.java]: thanks @gmunozfe

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
@mareknovotny this is ready to merge

**Review by mareknovotny — APPROVED** [review]

**Pull request merged**